error_check.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * Checks if required fields are selected
+ *
+ * @param data array containing request items
+ * @param need_data If request has a data section
+ *
+ * @throws No_URL_Select If no URL is present
+ * @throws No_Data If request requires data but none passed in
+ */
+function error_check($data, $need_data){
+
+	if (!isset($data["url"])){
+		$error = "No URL Selected";
+		throw new Exception($error);
+	}
+	if ($need_data){
+		if (!isset($data["data"])){
+			$error = "No Data";
+			throw new Exception($error);
+		}
+	}
+}
+
+/**
+ * Split url based on /
+ *
+ * @param url url for request
+ *
+ * @return associative array with keys as database and values as ids
+ */
+function parse($url){
+    // Splitting string by /
+    $url_arr = preg_split("/\//", $url);
+
+    // If there are odd number of parameters, the max even number of parameters
+    // are stored in the associative array. The remaining one will be added
+    // later and the value will be set to null.
+    $odd = FALSE;
+    if (sizeof($url_arr)%2==0){
+        $max = sizeof($url_arr);
+    }else{
+        $max = sizeof($url_arr)-1;
+        $odd = TRUE;
+    }
+
+    $parsed_url = array();
+    for ($index = 0; $index<$max; $index+=2){
+        $parsed_url[$url_arr[$index]] = $url_arr[$index+1];
+    }
+
+    if ($odd){
+        $parsed_url[$url_arr[$max]] = NULL;
+    }
+
+    return $parsed_url;
+}
+
+/**
+* Parse url to get an associative array with table and condition for operation
+* on table
+* @return returns an associative array that provies the table and condition
+* to operate on
+*/
+function url_to_params($url){
+	$result = array('table' => NULL, 'condition' => "");
+	$i=0;
+	foreach ($url as $key => $value) {
+		if($value != NULL){
+			switch($key){
+				case 'users':
+					$result['condition'] .= "utorid = '$value' AND ";
+					break;
+				case 'courses':
+					$result['condition'] .= "course_code = '$value' AND ";
+					break;
+				case 'sections':
+					$result['condition'] .= "section_id = $value AND ";
+					break;
+				case 'survey':
+					$result['condition'] .= "id = $value AND ";
+					break;
+				case 'responses':
+					$result['condition'] .= "response_id = $value AND ";
+					break;
+				default:
+					$error = "invalid url key";
+					throw new Exception($error);
+			}
+		}
+		++$i;
+		if($i == count($url)){
+			$result['table'] = $key;
+		}
+	}
+	$result['condition'] = substr($result['condition'], 0,-4);
+	return $result;
+}
+
+/**
+ * Checks if column exists within database.
```

Comment:
Say that you check whether the given table exists, the given columns are in that table, and (at least for now) whether the operation is allowed on that table.